### ISSUE TITLE
"Emergency mode" plugin?

### ISSUE THREAD

**Issue by wolfboyft** [DFHack/dfhack#2343]
Forces your units to always run while it's on. Or maybe select units saved with persist-table, though I don't know if that is exposed to C++. I guess it could be a script anyway.

**Comment by wolfboyft** [comment]
"Always run" as in, up their speed. Sprint to construct the wall blocking off the horde or whatever. And by "your units" I meant your citizens.

**Comment by lethosor** [comment]
Oh, so like fastdwarf but with more control over the speed?

**Comment by wolfboyft** [comment]
You could say that, but I don't think there's any vanilla-appropriate way to make them work faster like there is for moving faster. Moving faster on demand is a feature that already exists in adventure mode and should do in fortress mode.

**Comment by wolfboyft** [comment]
Oh that would be cool, yeah. How about we make this script/plugin and then get siren to invoke it?

**Comment by myk002** [comment]
why not write it as *part* of the siren script? I'm not sure if we need another top-level script name to add to our list. I'd like to start consolidating functionality into fewer scripts with more features each.

**Comment by wolfboyft** [comment]
Can we at least expose it from siren for custom scripts? I feel like I would want to use emergency running mode without siren but can see why more top level scripts would be a problem.

**Comment by myk002** [comment]
Absolutely. Any script can provide an API. I suppose there's a case for putting the code in modtools/ too and siren calling it.

As a side note, I'm not really a fan of modtools scripts being callable as commands. I feel they should just be APIs in library/lua/modtools/

**Comment by wolfboyft** [comment]
Totally agree on modtools being better off as an API.
What is the method for defining and using an API in a script that can also be called?

**Comment by myk002** [comment]
You just define a global function. See for example quickfort.lua (with API docs at https://docs.dfhack.org/en/stable/docs/tools/quickfort.html#api)

**Comment by lethosor** [comment]
> You could say that, but I don't think there's any vanilla-appropriate way to make them work faster like there is for moving faster. Moving faster on demand is a feature that already exists in adventure mode and should do in fortress mode.

"vanilla-appropriate" is heavily dependent on the player. I would personally classify any speedups as cheaty, aside from maybe `timestream`.

If the goal is to make units do _some things_ faster (e.g. movement but not jobs), I think that still fits under `fastdwarf` better than under `siren`. If some fastdwarf logic is moving to an API, that could enable siren to use it.

**Comment by wolfboyft** [comment]
Oh, I was assuming I would be able to set the running speed somewhere like what is set in adventure mode, which would not use the fastdwarf API, but rather would let the game tire out the units from sprinting and make them sweat etc. Now that I think about it though it could get unset back to walking between my setting it and the unit actually making a move action...
